packages/gatsby/src/commands/build.ts

```diff
@@ -193,6 +193,7 @@ module.exports = async function build(
           rootDir: program.directory,
           components: state.components,
           staticQueriesByTemplate: state.staticQueriesByTemplate,
+          webpackCompilationHash: webpackCompilationHash ?? ``,
```

Comment:
cast `as string` instead of fallback (maybe add comment why TS doesn't know it should be just `string` at this point

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/34225/commits/30ba3250623606ee77b35f2fdff3e6358b3a827d